src/main/java/com/thirdcc/webapp/authorization/ManagementTeamSecurityExpression.java

```diff
@@ -0,0 +1,100 @@
+    private final UserRepository userRepository;
+    private final EventCrewRepository eventCrewRepository;
+    private final AdministratorRepository administratorRepository;
+    private final YearSessionRepository yearSessionRepository;
+
+
+    public ManagementTeamSecurityExpression(
+        TokenProvider tokenProvider,
+        UserRepository userRepository,
+        EventCrewRepository eventCrewRepository,
+        AdministratorRepository administratorRepository,
+        YearSessionRepository yearSessionRepository
+    ) {
+        this.tokenProvider = tokenProvider;
+        this.userRepository = userRepository;
+        this.eventCrewRepository = eventCrewRepository;
+        this.administratorRepository = administratorRepository;
+        this.yearSessionRepository = yearSessionRepository;
+    }
+
+    public boolean isCurrentCCHead() {
+        User currentUser = getCurrentUserWithLogin();
+        YearSession currentYearSession = yearSessionRepository.findFirstByOrderByIdDesc()
+            .orElseThrow(() -> new BadRequestException("Year Session not found"));
+        Administrator administrator = administratorRepository
+            .findByUserIdAndYearSessionAndRole(currentUser.getId(), currentYearSession, AdministratorRole.CC_HEAD)
+            .orElse(null);
+        return administrator != null;
+    }
+
+    public boolean hasRoleAdminOrIsEventCrew(Long eventId) {
+        if (getUserAuthRole().contains(AuthoritiesConstants.ADMIN)) {
+            return true;
+        } else {
+            return isEventCrew(eventId);
+        }
+    }
+
+    public boolean isEventCrew(Long eventId) {
+        User currentUser = getCurrentUserWithLogin();
+        EventCrew eventCrew = eventCrewRepository
+            .findByUserIdAndAndEventId(currentUser.getId(), eventId)
+            .orElse(null);
+        return eventCrew != null;
+    }
+
+    public boolean isEventHead(Long eventId) {
+        User currentUser = getCurrentUserWithLogin();
+        EventCrew eventCrew = eventCrewRepository
+            .findByUserIdAndAndEventId(currentUser.getId(), eventId)
+            .orElse(null);
+        return eventCrew != null && eventCrew.getRole().equals(EventCrewRole.HEAD);
+    }
+
+    private List<String> getUserAuthRole() {
+        String currentUserJWT = SecurityUtils
+            .getCurrentUserJWT()
+            .orElseThrow(() -> new BadRequestException("User is not login"));
+        Jws<Claims> claimsJws = tokenProvider
+            .getJWT()
+            .parseClaimsJws(currentUserJWT);
+        return Arrays
+            .stream(claimsJws.getBody().get("auth").toString().split(","))
+            .collect(Collectors.toList());
+    }
+
+    private User getCurrentUserWithLogin() {
+        return SecurityUtils
+            .getCurrentUserLogin()
+            .flatMap(userRepository::findOneWithAuthoritiesByLogin)
+            .orElseThrow(() -> new BadRequestException("Cannot find user"));
+    }
```

Comment:
actually we can use 
`userService.getUserWithAuthorities()`
instead of decode the token ourselves

Reply:
@kachun333 feel free to check with new commit
56b8ebb76224c0ad2f5496cb06dcc7738dc9dfff